### ISSUE TITLE
Script for serving the model ie it does search, document,refine, answer on its own based on the initial prompt

### ISSUE THREAD

**Issue by prasadke20** [syr-cn/AutoRefine#2]
Hi, great work, is there any script to test the model for some input prompts, similar script exists in Search-R1 repo with name infer.py

**Comment by syr-cn** [comment]
Thanks for your interest and kind reminder! We will add an `infer.py` for better utilization of this work soon. I will update in issue once it comes out.

**Comment by syr-cn** [comment]
Hi @prasadke20, we've added `infer.py` and `demo.py` for fast inference using our code. The instructions about how to use these scripts are included in `README.md`

If you have further suggestions, please let me know :)

**Comment by prasadke20** [comment]
I have one more suggestion, where it would be great if you could tell how to train on your own data, currently i went through the reward manager and compute_em functions to get an idea whats happening, a readme about how to train on our own data would be a great add on.

**Comment by syr-cn** [comment]
Thanks for your suggestion!

To train on your own data, you may modify the code [here](http://github.com/syr-cn/AutoRefine/blob/main/preprocess/data_process/qa_search_train_merge.py#L51):
```python
dataset = datasets.load_dataset('RUC-NLPIR/FlashRAG_datasets', data_source)

train_dataset = dataset['train']
# train_dataset = train_dataset.shuffle(seed=42).select(range(100))

# add a row to each data item that represents a unique id
def make_map_fn(split):

    def process_fn(example, idx):
        example['question'] = example['question'].strip()
        if example['question'][-1] != '?':
            example['question'] += '?'
        question = make_prefix(example, template_type=args.template_type)
        solution = {
            "target": example['golden_answers'],
        }

        data = {
            "data_source": data_source,
            "prompt": [{
                "role": "user",
                "content": question,
            }],
            "ability": "fact-reasoning",
            "reward_model": {
                "style": "rule",
                "ground_truth": solution
            },
            "extra_info": {
                'split': split,
                'index': idx,
            }
        }
        return data
```

We will notice you once we have the related updates in the docs!

**Comment by prasadke20** [comment]
Thanks @syr-cn
